### PR TITLE
Add getSelectedCivZone to dfhack.gui

### PR DESCRIPTION
**Pull request by kelvie**
Mostly helpful for my own automation in my custom lua scripts.

Tested it with barracks, meeting areas, and pastures.

image

### CODE REVIEW

File: library/modules/Gui.cpp [hunk 1]
```diff
@@ -1111,6 +1112,26 @@ df::building_stockpilest* Gui::getSelectedStockpile(color_ostream& out, bool qui
...
 }
 
+bool Gui::any_civzone_hotkey(df::viewscreen* top) {
+    return getAnyCivZone(top) != NULL;
+}
+
+df::building_civzonest *Gui::getAnyCivZone(df::viewscreen* top) {
+    if (matchFocusString("dwarfmode/Zone")) {
+        return game->main_interface.civzone.cur_bld;
+    }
```
Comment (myk002) [inline library/modules/Gui.cpp:1122]: Just to check, does this get the right value when multiple zones overlap?

Comment (kelvie) [reply library/modules/Gui.cpp:1122]: I shall check tonight, I more or less copy and pasted this logic from the other `get` functions.

Comment (myk002) [reply library/modules/Gui.cpp:1122]: ack. the reason I'm suspicious is that civzones are the only building type that can overlap, so the rules might be different.

Comment (kelvie) [reply library/modules/Gui.cpp:1122]: Just tested, this works properly when they're completely overlapped (tested a meeting hall and an office stacked on top of each other exactly, when I change the selection using the left and right arrows here:

image

the function returns the correct civzone.

**Review by myk002 — COMMENTED** [review]
could you add changelog entries under API for the cpp code and under Lua for the Lua function?

**Comment by kelvie** [comment]
> could you add changelog entries under API for the cpp code and under Lua for the Lua function?

Oh, yeah, didn't realize these were documented. I'll do this tomorrow if I have time.

**Comment by kelvie** [comment]
Updated dogs and changelog, waiting for build to check formatting.

**Review by myk002 — APPROVED** [review]

**Pull request merged**